src/sw.ts

```diff
@@ -14,40 +14,43 @@
  *  limitations under the License.
  */
 
-const workboxAddr = 'https://storage.googleapis.com/workbox-cdn/releases/6.2.0/workbox-sw.js'; // TODO: host this file in this project?
-importScripts(workboxAddr);
+import { registerRoute } from 'workbox-routing';
+import { StaleWhileRevalidate } from 'workbox-strategies';
 
 const cacheName = 'app-cache';
 
 const mainDBName = 'appDB';
 const objStoreName = 'imageStore';
 const mainImageName = 'mainImage';
 
-self.addEventListener('install', function (event) {
+interface InstallEvent extends Event {
```

Comment:
Declaring `self` as the right interface will probably remove the need for this:

```
declare var self: ServiceWorkerGlobalScope;
```

Reply:
done